virttest/utils_misc.py

```diff
@@ -2750,20 +2750,37 @@ def format_linux_disk(session, did, mountpoint, size, fstype="ext3"):
     :param ostype: guest os type 'windows' or 'linux'.
     :return Boolean: disk usable or not.
     """
-    list_disk_cmd = "lsblk -p -oKNAME,MOUNTPOINT,SERIAL,SIZE,MODEL,WWN"
+    list_disk_cmd = "lsblk -oKNAME,MOUNTPOINT,SIZE,MODEL"
     disks = session.cmd_output(list_disk_cmd)
     logging.debug("Disks detail: %s" % disks)
     for disk in disks.splitlines():
-        if did in disk:
-            kname = disk.split()[0]
-            logging.info("Create partition on disk '%s'" % kname)
-            mkpart_cmd = "parted -s %s  mkpart primary 0 %s" % (kname, size)
+        kname = disk.split()[0]
+        get_disk_info_cmd = "udevadm info -q property -p /sys/block/%s" % kname
+        output = session.cmd_output(get_disk_info_cmd)
+        re_str = r"(?<=DEVNAME=/dev/)(.*)|(?<=ID_SERIAL=)(.*)| \
+                 (?<=ID_SERIAL_SHORT=)(.*)|(?<=ID_WWN=)(.*)"
+        did_list_group = re.finditer(re_str, output, re.M)
+        did_list = []
+        for i in did_list_group:
+            did_list.append(i.group())
+
+        if did in did_list:
+            if size is None:
+                size = disk.split()[1]
+            devname = os.path.join("/dev", kname)
+            logging.info("Create partition on disk '%s'" % devname)
+            mkpart_cmd = "parted -s %s mklabel gpt mkpart primary 0 %s" % \
+                         (devname, size)
             session.cmd(mkpart_cmd)
+            partname = devname + "1"
```

Comment:
start size in line#2772 is always '0', so only one partition support in the disk.